test/dlc_tests/sanity/test_security_check.py

```diff
@@ -64,24 +81,61 @@ def test_ecr_scan(image, ecr_client, sts_client, region):
         _, image_repo_name = image_repo_uri.split("/")
         target_image_repo_name = f"beta-{image_repo_name}"
         image = ecr_utils.reupload_image_to_test_ecr(image, target_image_repo_name, region)
-    minimum_sev_threshold = "HIGH"
-    scan_status = None
-    start_time = time()
-    ecr_utils.start_ecr_image_scan(ecr_client, image)
-    while (time() - start_time) <= 600:
-        scan_status, scan_status_description = ecr_utils.get_ecr_image_scan_status(ecr_client, image)
-        if scan_status == "FAILED" or scan_status not in [None, "IN_PROGRESS", "COMPLETE"]:
-            raise RuntimeError(scan_status_description)
-        if scan_status == "COMPLETE":
-            break
-        sleep(1)
-    if scan_status != "COMPLETE":
-        raise TimeoutError(f"ECR Scan is still in {scan_status} state. Exiting.")
-    severity_counts = ecr_utils.get_ecr_image_scan_severity_count(ecr_client, image)
-    scan_results = ecr_utils.get_ecr_image_scan_results(ecr_client, image, minimum_vulnerability=minimum_sev_threshold)
-    assert all(
-        count == 0 for sev, count in severity_counts.items() if CVESeverity[sev] >= CVESeverity[minimum_sev_threshold]
-    ), (
-        f"Found vulnerabilities in image {image}: {str(severity_counts)}\n"
-        f"Vulnerabilities: {json.dumps(scan_results, indent=4)}"
+
+    run_scan(ecr_client, image)
+    scan_results = ecr_utils.get_ecr_image_scan_results(ecr_client, image, minimum_vulnerability=MINIMUM_SEV_THRESHOLD)
+
+    image_scan_allowlist = ScanVulnerabilityList(minimum_severity=CVESeverity[MINIMUM_SEV_THRESHOLD])
+    image_scan_allowlist_path = get_ecr_scan_allowlist_path(image)
+    if os.path.exists(image_scan_allowlist_path):
+        image_scan_allowlist.construct_allowlist_from_file(image_scan_allowlist_path)
+
+    ecr_image_vulnerability_list = ScanVulnerabilityList(minimum_severity=CVESeverity[MINIMUM_SEV_THRESHOLD])
+    ecr_image_vulnerability_list.construct_allowlist_from_ecr_scan_result(scan_results)
+
+    remaining_vulnerabilities = ecr_image_vulnerability_list - image_scan_allowlist
+
+    assert not remaining_vulnerabilities, (
```

Comment:
Is the motive of the ECR Scan tests to just check if new vulnerabilities have been found on a given image? Or, do the tests also check if security patches for the existing vulnerabilities have been released?